hbase-client/src/test/java/org/apache/hadoop/hbase/quotas/TestRpcThrottlingException.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.hadoop.hbase.quotas;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
+import org.apache.hadoop.hbase.HBaseClassTestRule;
+import org.apache.hadoop.hbase.testclassification.SmallTests;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+
+@Category({ SmallTests.class })
+public class TestRpcThrottlingException {
+
+  @ClassRule
+  public static final HBaseClassTestRule CLASS_RULE =
+    HBaseClassTestRule.forClass(TestRpcThrottlingException.class);
+
+  @Test
+  public void itHandlesSecWaitIntervalMessage() {
+    try {
+      RpcThrottlingException.throwNumReadRequestsExceeded(1001);
+      fail();
+    } catch (RpcThrottlingException e) {
+      assertTrue(e.getMessage().contains("wait 1sec, 1ms"));
+    }
+  }
+
+  @Test
+  public void itHandlesMsWaitIntervalMessage() {
+    try {
+      RpcThrottlingException.throwNumReadRequestsExceeded(50);
+      fail();
+    } catch (RpcThrottlingException e) {
+      assertTrue(e.getMessage().contains("wait 50ms"));
+    }
+  }
+
+  @Test
+  public void itHandlesMinWaitIntervalMessage() {
+    try {
+      RpcThrottlingException.throwNumReadRequestsExceeded(65_015);
+      fail();
+    } catch (RpcThrottlingException e) {
+      assertTrue(e.getMessage().contains("wait 1mins, 5sec, 15ms"));
+    }
+  }
+
+  @Test
+  public void itConvertsMillisToString() {
+    String output = RpcThrottlingException.stringFromMillis(6500);
+    assertEquals("6sec, 500ms", output);
+  }
+
+  @Test
+  public void itConvertsStringToMillis() {
+    long millis = RpcThrottlingException.timeFromString("5mins, 2sec, 0ms");
```

Comment:
can you make a few more tests? sorry.  TBH I might just use test inputs like this to keep it simple:

```
// new format
"1hrs, 1mins, 1sec, 1ms"
"1mins, 1sec, 1ms"
"1sec, 1ms"
"1ms"
// legacy format
"1hrs, 1mins, 1sec"
"1mins, 1sec"
"1sec"
```

This way we can be sure we're actually parsing the values correctly -- in this test i commented on, using 0ms proves that the match works but doesn't prove that the parsing of value actually works since the value is 0.